debot/src/main/java/com/tomoima/debot/DebotDialog.kt

```diff
@@ -0,0 +1,72 @@
+package com.tomoima.debot
+
+import android.content.DialogInterface
+import android.os.Bundle
+import android.support.annotation.VisibleForTesting
+import android.support.v4.app.DialogFragment
+import android.support.v4.app.FragmentActivity
+import android.support.v4.app.FragmentManager
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.AdapterView
+import android.widget.ListView
+import com.tomoima.debot.adapter.DebotMenuListAdapter
+
+// This class allows others to inherit
+open class DebotDialog : DialogFragment() {
+    private val TAG = "com.tomoima.debot.Debot"
+    private val debotStrategyList = DebotStrategies.get()
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        isCancelable = true
+        onCancel(object : DialogInterface {
+            override fun cancel() {
+                dismissDebugMenu()
+            }
+
+            override fun dismiss() {
+                dismissDebugMenu()
+            }
+        })
+    }
+
+
+    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
+        val view = inflater.inflate(R.layout.fragment_debot, container)
+        val list = view.findViewById(R.id.debug_menu_list) as ListView
+        val adapter = DebotMenuListAdapter(activity, debotStrategyList)
+        list.adapter = adapter
+        list.onItemClickListener = AdapterView.OnItemClickListener { _, _, position, _ ->
+            debotStrategyList[position].startAction(activity)
+            dismissDebugMenu()
+        }
+
+        return view
+    }
+
+
+    fun showDebugMenu(activity: FragmentActivity?) {
+        val fragmentManager = activity?.supportFragmentManager as? FragmentManager
+        if (fragmentManager != null && fragmentManager.findFragmentByTag(TAG) == null) {
+                show(fragmentManager, TAG)
+                fragmentManager.executePendingTransactions()
+        }
+    }
+
+    fun dismissDebugMenu() {
+        activity.let { dismissDebugMenu(it.supportFragmentManager) }
```

Comment:
if `activity` is `nullable`, you should use `?.let`
```kotlin
activity?.let { dismissDebugMenu(it.supportFragmentManager) }
```
From the documentation ( https://kotlinlang.org/api/latest/jvm/stdlib/kotlin/let.html )
> Calls the specified function block with this value as its argument and returns its result.

the `let` method says "run this code on on this object". This can still still throw a `NullPointerException` if we call it straight on the object, because the let block is implemented as `block(this)`

but if we use `?.let`, it says "run the following method **if the object is not null**", and that is where we get null safety from